### ISSUE TITLE
second PCA analysis on the same TrajectoryIterator leads to error

### ISSUE THREAD

**Issue by hongbo-zhu-cn** [Amber-MD/pytraj#1452]
suppose you have a `TrajectoryIterator` instance `traj` and you apply pca analysis with `fit=True`:

```
import pytraj as pt
pcadata1 =  pt.pca(traj, mask=':1-10@CA', fit=True, ref=0)  
# then run the second pca analysis on the same traj (e.g you perhaps want to apply different masks/ref)
pcadata2 =  pt.pca(traj, mask=':11-20@CA', fit=True, ref=1)  
```
It will give you error:
```
AttributeError: 'NoneType' object has no attribute 'box'
```

I locate the error to the function `TrajectoryCpptraj::_remove_transformations` in file: `./pytraj/trajectory/c_traj/c_trajectory.pyx`. The `self._actionlist` is initialized without correctly specifying topology:
```
self._actionlist = ActionList()
```
should be 
```
self._actionlist = ActionList(top=self.top)
```

Actually, I worry that even after the fix, the code will still leak memory. So it is perhaps better to replace the first two initialization lines 
```
self._actionlist = ActionList()
self._cdslist = CpptrajDatasetList()
```
with 
```
self._initialize_actionlist()
```
So that memory is properly freed after each pca analysis.

See my branch of the code for the fix proposal.

**Comment by hainm** [comment]
Thanks. Would you mind making a pull request and add a test for it?

```
diff --git a/tests/test_analysis/test_pca.py b/tests/test_analysis/test_pca.py
index d1c412d..a2e6e28 100644
--- a/tests/test_analysis/test_pca.py
+++ b/tests/test_analysis/test_pca.py
@@ -228,6 +228,12 @@ class TestPCA(unittest.TestCase):
         data1, _ = pt.pca(traj_on_mem, mask='@CA', n_vecs=2, fit=fit, ref=ref1)
         aa_eq(np.abs(data0), np.abs(data1))
 
+        # try again
+        # https://github.com/Amber-MD/pytraj/issues/1452
+        data2, _ = pt.pca(
+            traj_on_disk, mask='@CA', n_vecs=2, fit=fit, ref=ref0)
+        aa_eq(np.abs(data0), np.abs(data2))
+
     def test_traj_on_disk_fit_to_given_reference_and_restore_transform_commands(
             self):
         """test_traj_on_disk_fit_to_given_reference_and_restore_transform_commands
```

**Comment by hongbo-zhu-cn** [comment]
I just did it. Would you have a look? thx

**Comment by hainm** [comment]
hi, I made a comment there. Can you address it? Thanks.

**Comment by hongbo-zhu-cn** [comment]
good idea. Just added the comparison.

**Comment by hainm** [comment]
closed by #1453
